sympy/functions/combinatorial/numbers.py

```diff
@@ -1858,12 +1858,27 @@ def stirling(n, k, d=None, kind=2, signed=False):
 
 
 @cacheit
-def _nT(n, k):
+def _nT(n, k=None):
     """Return the partitions of ``n`` items into ``k`` parts. This
     is used by ``nT`` for the case when ``n`` is an integer."""
-    if k == 0:
-        return 1 if k == n else 0
-    return sum(_nT(n - k, j) for j in range(min(k, n - k) + 1))
+    if k is None:
```

Comment:
eliminate this since the calling routine knows the value of k and no user should be calling this directly unless they have sanitized their data ... and know what k is ;-)